packages/core/src/rum/instrumentation/resourceTracking/DdRumResourceTracking.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+import { InternalLog } from '../../../InternalLog';
+import { SdkVerbosity } from '../../../SdkVerbosity';
+
+import { firstPartyHostsRegexBuilder } from './domain/firstPartyHosts';
+import type { RequestProxy } from './domain/interfaces/RequestProxy';
+import { XHRProxy } from './implementation/XHRProxy';
+
+/**
+ * Provides RUM auto-instrumentation feature to track resources (fetch, XHR, axios) as RUM events.
+ */
+export class DdRumResourceTracking {
+    private static isTracking = false;
+    private static requestProxy: RequestProxy;
+
+    /**
+     * Starts tracking resources and sends a RUM Resource event every time a network request is detected.
+     */
+    static startTracking({
+        tracingSamplingRate,
+        firstPartyHosts
+    }: {
+        tracingSamplingRate: number;
+        firstPartyHosts: string[];
+    }): void {
+        // extra safety to avoid proxying the XHR class twice
+        if (DdRumResourceTracking.isTracking) {
+            InternalLog.log(
+                'Datadog SDK is already tracking XHR resources',
+                SdkVerbosity.WARN
+            );
+            return;
+        }
+
+        this.requestProxy = new XHRProxy(XMLHttpRequest);
+        this.requestProxy.onTrackingStart({
+            tracingSamplingRate,
+            firstPartyHostsRegex: firstPartyHostsRegexBuilder(firstPartyHosts)
+        });
+
+        InternalLog.log(
+            'Datadog SDK is tracking XHR resources',
+            SdkVerbosity.INFO
+        );
+        DdRumResourceTracking.isTracking = true;
+    }
+
+    static stopTracking(): void {
+        if (DdRumResourceTracking.isTracking) {
+            DdRumResourceTracking.isTracking = false;
+            this.requestProxy.onTrackingStop();
```

Comment:
should we set `requestProxy` to `null` here as well? Just wondering if there is any memory leak possible by keeping the stuff in the `static` field.

Reply:
that is a good idea, I think it is a good practice to set, so that we don't have to think about memory leaks